### ISSUE TITLE
Media async: ConcurrentModificationException

### ISSUE THREAD

**Issue by mzorz** [wordpress-mobile/WordPress-Android#6258]
```
07-07 15:31:44.561 1698-1698/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: main
                                                                     org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
                                                                         at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:505)
                                                                         at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:487)
                                                                         at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:479)
                                                                         at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                         at android.os.Looper.loop(Looper.java:137)
                                                                         at android.app.ActivityThread.main(ActivityThread.java:5041)
                                                                         at java.lang.reflect.Method.invokeNative(Native Method)
                                                                         at java.lang.reflect.Method.invoke(Method.java:511)
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560)
                                                                         at dalvik.system.NativeStart.main(Native Method)
                                                                      Caused by: java.util.ConcurrentModificationException
                                                                         at java.util.HashMap$HashIterator.nextEntry(HashMap.java:792)
                                                                         at java.util.HashMap$KeyIterator.next(HashMap.java:819)
                                                                         at org.wordpress.android.ui.media.services.MediaUploadService.updatePostAndStopServiceIfUploadsComplete(MediaUploadService.java:330)
                                                                         at org.wordpress.android.ui.media.services.MediaUploadService.uploadNextInQueue(MediaUploadService.java:192)
                                                                         at org.wordpress.android.ui.media.services.MediaUploadService.handleOnMediaUploadedSuccess(MediaUploadService.java:137)
                                                                         at org.wordpress.android.ui.media.services.MediaUploadService.onMediaUploaded(MediaUploadService.java:396)
                                                                         at java.lang.reflect.Method.invokeNative(Native Method)
                                                                         at java.lang.reflect.Method.invoke(Method.java:511)
                                                                         at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:485)
                                                                         at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:479) 
                                                                         at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67) 
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                         at android.os.Looper.loop(Looper.java:137) 
                                                                         at android.app.ActivityThread.main(ActivityThread.java:5041) 
                                                                         at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                         at java.lang.reflect.Method.invoke(Method.java:511) 
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560) 
                                                                         at dalvik.system.NativeStart.main(Native Method) 
07-07 15:31:44.561 380-390/system_process W/ActivityManager:   Force finishing activity org.wordpress.android/.ui.main.WPMainActivity
07-07 15:31:44.641 380-383/system_process D/dalvikvm: GC_CONCURRENT freed 975K, 23% free 13064K/16788K, paused 1ms+2ms, total 9ms
````

**Comment by kwonye** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [wtf](https://trello.com/c/L5vWVPnE/3-wtf)

**Comment by mzorz** [comment]
Closed via #6259
